### DISCUSSION TITLE
Configurable input and output models

### DISCUSSION THREAD

**Discussion by ZexDC** [danielgtaylor/huma#681] [Q&A]
Hello, I've been using Go structs to define inputs and outputs and it has been working great! I was now thinking of a more customizable approach.

Is there any way to load in query fields, from a file for example, and generate input/output models from it?
Ideally just by adding extra query fields to the file (specifying field name, tag, type, validation, doc), Huma would generate the complete schema for all the fields.
This probably requires the use of maps instead of structs, but from what I understood Huma only expects structs, unless I missed something from the documentation?

**Comment by danielgtaylor** [comment]
You could try building a code generator (look into Golang code generation, it's fairly straightforward) that outputs structs with tags from some input format you create. There's not really a programmatic way to create structs at runtime that you could write code against to do something with the values you get.

**Reply by ZexDC** [reply]
Thanks, yeah I realized there are workarounds to generate structs at runtime, but not really recommended. 
Would it be too much against Huma's design to add dynamic maps as alternative to structs for query parameters?
